File: backend/src/main/java/ru/sportexpress/menuApp/web/service/MenuServiceImpl.java
package ru.sportexpress.menuApp.web.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.sportexpress.menuApp.entity.Menu;
import ru.sportexpress.menuApp.exceptions.MenuNotFoundException;
import ru.sportexpress.menuApp.repository.MenuRepository;

import java.util.List;
import java.util.Optional;

@Service
public class MenuServiceImpl implements MenuService {

    @Autowired
    private MenuRepository menuRepository;

    @Override
    public List<Menu> getAll() {


        return (List<Menu>) menuRepository.findAll();
    }

    @Override
    public Optional<Menu> getOne(Integer id) {
        return menuRepository.findById(id);
    }

    @Override
    public Menu update(int id, Menu menu) {
        Menu menu1 = menuRepository.findById(id).orElse(null);
        if (menu1 == null) {
            throw new MenuNotFoundException("not found menu with id "+ id);
        }
        menu1.setName(menu.getName());
        menu1.setLink(menu.getLink());
        menu1.setShortLink(menu.getShortLink());
        menuRepository.save(menu1);
        return menu1;
    }

    @Override
    public Menu save(Menu menu) {
        Menu menu1 = menuRepository.save(menu);
        return menu1;
    }

    @Override
    public Menu delete(int id) {
        Menu menu2 = menuRepository.findById(id).orElse(null);
        if (menu2 == null) {
            throw new MenuNotFoundException("not found menu with id "+ id);
        }
        menuRepository.delete(menu2);
        return null;
    }
}

File: backend/src/main/kotlin/ru/sportexpress/menuApp/web/service/MenuServiceImpl.kt
package ru.sportexpress.menuApp.web.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import ru.sportexpress.menuApp.entity.Menu
import ru.sportexpress.menuApp.exceptions.ResourceNotFoundException
import ru.sportexpress.menuApp.repository.MenuRepository
import ru.sportexpress.menuApp.requests.MenuResponse
import ru.sportexpress.menuApp.requests.Status
import java.util.*


@Service
class MenuServiceImpl : MenuService {

    @Autowired
    lateinit var menuRepository: MenuRepository

    override fun getAll(): MenuResponse {
        val menuResponse = MenuResponse()
        menuResponse.status = Status.OK
        menuResponse.result = menuRepository.findAllByParentIdIsNull()
        return menuResponse
    }

    override fun getOne(id: Long): Optional<Menu> {
        return menuRepository.findById(id)
    }

    override fun update(id: Long, menu: Menu): Menu {
        val menu1: Menu = menuRepository.findById(id).orElseThrow { ResourceNotFoundException("Menu", "id", id) }
        menu1.title = menu.title
        menu1.linkUrl = menu.linkUrl
        menu1.active = menu.active
        menu1.parentId = menu.parentId
        menu1.weight = menu.weight
        menu1.sortType = menu.sortType
        menu1.parentId = menu.parentId
        menuRepository.save(menu1)
        return menu1
    }

    override fun save(menu: Menu): Menu {
        return menuRepository.save(menu)
    }

    override fun delete(id: Long): ResponseEntity<*> {
        val menu2: Menu = menuRepository.findById(id).orElseThrow { ResourceNotFoundException("Menu", "id", id) }
        menuRepository.delete(menu2)
        return ResponseEntity.ok().build<Any>()
    }

}
